core/arch/arch.c

```diff
@@ -3825,21 +3825,48 @@ dump_mcontext(priv_mcontext_t *context, file_t f, bool dump_xml)
     }
 #elif defined(AARCHXX)
     {
-        int i, j;
 #    ifdef AARCH64
-        int words = proc_has_feature(FEATURE_SVE) ? 16 : 4;
+        const uint elements = proc_has_feature(FEATURE_SVE)
+            ? (proc_get_vector_length_bytes() / sizeof(uint))
+            : 4;
+        /* XXX: should be proc_num_simd_saved(). */
+        const uint num_simd_regs = MCXT_NUM_SIMD_SVE_SLOTS;
+        const char *reg_prefix = proc_has_feature(FEATURE_SVE) ? "z" : "q";
 #    else
-        int words = 4;
-#    endif
+        const uint elements = 4;
         /* XXX: should be proc_num_simd_saved(). */
-        for (i = 0; i < proc_num_simd_registers(); i++) {
-            print_file(f, dump_xml ? "\t\tqd= \"0x" : "\tq%-3d= 0x", i);
-            for (j = 0; j < words; j++) {
+        const uint num_simd_regs = proc_num_simd_registers();
+        const char *reg_prefix = "q";
+#    endif
+        ASSERT(elements <=
+               sizeof(context->simd[0].u32) / sizeof(context->simd[0].u32[0]));
+        for (uint i = 0; i < num_simd_regs; i++) {
+            print_file(f, dump_xml ? "\t\t%s%d= \"0x" : "\t%s%-3d= 0x", reg_prefix, i);
+            for (uint j = 0; j < elements; j++) {
                 print_file(f, "%08x ", context->simd[i].u32[j]);
             }
             print_file(f, dump_xml ? "\"\n" : "\n");
         }
-        /* TODO i#5365: SVE predicate registers and FFR dump. */
+#    ifdef AARCH64
+        if (proc_has_feature(FEATURE_SVE)) {
+            /* Dump SVE P registers */
+            const uint pred_u16_elements =
```

Comment:
I think s/_elements/_count/ is clearer (else sounds like an array of of values or sthg to me)

Reply:
Done.